### PR TITLE
Initial Terraform for Demo cluster

### PR DESCRIPTION
**Pull request by williamdanger**
Check it out: [http://demo-ecs-ALB-1300727642.us-east-1.elb.amazonaws.com](demo-ecs-ALB-1300727642.us-east-1.elb.amazonaws.com)

**Lay of the land:**
- Everything's in the 'infra' directory
- Includes a Makefile with some common targets (plan, apply, etc). These targets rely on [aws-vault](https://github.com/99designs/aws-vault) being installed. Also makes CI easier when we trust this enough to run on merge (maybe in a separate 
- Structure is pretty flat but we can pull stuff into modules for more DRYness

Still need to setup Cloudflare DNS

### CODE REVIEW

File: infra/ecs-cluster.tf [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  name        = var.environment
+  description = "Private service discovery for ${var.environment} ECS cluster."
+  vpc         = module.vpc.vpc_id
+}
+
+// Log Group
+
+resource "aws_cloudwatch_log_group" "ecs" {
+  name              = "${var.environment}-${var.name}"
+  retention_in_days = var.cloudwatch_retention_days
```
Comment (pgebheim) [inline infra/ecs-cluster.tf:18]: Nice touch was just looking at this.

File: infra/service-0x.tf [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+    secretOptions = null
+  }
+  environment = [
+    {
+      name : "ETHEREUM_CHAIN_ID",
+      value : "42"
+    },
+    {
+      name : "P2P_ADVERTISE_ADDRS",
+      value : "/dns4/${var.environment}.${var.domain}/tcp/60559/ws/ipfs/16Uiu2HAmRMgvPQV2UYKXuuCnNaFLpc36PhLp2UKVcL1ePseVcz4y"
```
Comment (pgebheim) [inline infra/service-0x.tf]: Should probably make the public key (ipfs/<key>) an env as well.

File: infra/variables.tf [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+// Networking
+variable availability_zones {
+  description = "List of AZ used for provisioning network. Best practice to provide at least 2."
+}
+
+variable "vpc_cidr_block" {
+  description = "CIDR block for VPC"
+}
+
+variable "domain" {
```
Comment (pgebheim) [inline infra/variables.tf:34]: What's this ultimately used for? The root DNS for the ALB?

**Review by pgebheim — APPROVED** [review]
Very minor improvements can come in another PR

File: infra/variables.tf (hunk 3 above)
Comment (williamdanger) [reply infra/variables.tf:34]: It's used for looking up the SSL cert and for a few container configs (ex 0x). Also future-proof for when we manage DNS in code.

File: infra/ecs-cluster.tf (hunk 1 above)
Comment (williamdanger) [reply infra/ecs-cluster.tf:18]: There's definitely room for improvement. Getting some decent observability will help debugging.

**Comment by justinbarry** [comment]
This is really cool.

**Pull request merged**